### ISSUE TITLE
Possible type mismatch for t_comm parameter in C version of TRF5_1?

### ISSUE THREAD

**Issue by ksyx** [osdldbt/dbt5#15]
While running dbt5-run I encountered exceptions in TradeResult, but it appears that the error message of exception thrown in https://github.com/osdldbt/dbt5/blob/5575431ad9826c532a466191239338c635002b0f/src/transactions/pgsql/DBConnection.cpp#L218-L220 has already been freed in the catch block at
https://github.com/osdldbt/dbt5/blob/5575431ad9826c532a466191239338c635002b0f/src/BrokerageHouse/BrokerageHouse.cpp#L197-L203, leading `str` to be a garbage value.
Therefore I ran TestTxn with gdb and got the followings:
```
(gdb) p sql
$1 = 0x7fffe8009de0 "SELECT * FROM TradeResultFrame5(4300000001,19.54,'CMPT','2024-5-22 0:53:15',200000008728315,27.14)"
(gdb) printf "%s", (const char *)PQresultErrorMessage(res)
ERROR:  numeric field overflow
DETAIL:  A field with precision 10, scale 2 must round to an absolute value less than 10^8.
CONTEXT:  SQL statement "UPDATE trade
SET t_comm = $1
  , t_dts = $2
  , t_st_id = $3
  , t_trade_price = $4
WHERE t_id = $5"
```

This statement appears to be TRF5_1 and upon checking the implementation I found that all values related with `comm_amount` is processed using float8/double, and there is not another mention of FLOAT4OID in the repo. After changing it to FLOAT8OID the exception is no longer thrown, but I cannot figure out a reason, and I am also wondering whether there are any rationale behind using FLOAT4 specifically for this parameter? Thanks.

https://github.com/osdldbt/dbt5/blob/5575431ad9826c532a466191239338c635002b0f/scripts/pgsql/dbt5-pgsql-create-tables.in#L235
https://github.com/osdldbt/dbt5/blob/5575431ad9826c532a466191239338c635002b0f/scripts/pgsql/dbt5-pgsql-create-tables.in#L392

https://github.com/osdldbt/dbt5/blob/5575431ad9826c532a466191239338c635002b0f/storedproc/pgsql/c/trade_result.c#L208-L214

https://github.com/osdldbt/dbt5/blob/5575431ad9826c532a466191239338c635002b0f/storedproc/pgsql/c/trade_result.c#L1654
https://github.com/osdldbt/dbt5/blob/5575431ad9826c532a466191239338c635002b0f/storedproc/pgsql/c/trade_result.c#L1667-L1668

https://github.com/osdldbt/dbt5/blob/5575431ad9826c532a466191239338c635002b0f/storedproc/pgsql/c/trade_result.c#L367

![image](https://github.com/osdldbt/dbt5/assets/18738953/c0df727f-f835-496f-8e39-807e6197b176)

**Comment by markwkm** [comment]
Thanks for catching that.  The old C user defined functions need some TLC. :)

I believe float4 was used because S_PRICE_T was defined in a way that float4 should have been big enough.  I have seen in other publications that vendors are increasing the size of S_PRICE_T, and I tried to do the same in a couple places.  So I think it's right to try to make sure we are using float8....

**Comment by ksyx** [comment]
Thanks for confirming! Another thing I spotted, as in the beginning of this issue, is that the string pointed to `msg.str().c_str()` seemed to be freed while the exception is being thrown, and this makes the `str` value in the catch block unable to be used. Fixing this can be helpful as logging `str` in error message can save some effort making it to run with a debugger. So is it a good idea to create a new issue for this? Thanks

**Comment by markwkm** [comment]
I think has has been addressed in f240d86c810587035590ff48a4e8f3f0082dbe1b and e1ce4a7ad7b9a38e9d5029570bb1209c2a7907fc.
